ztp/kube-compare-reference/compare.sh

```diff
@@ -0,0 +1,46 @@
+#! /bin/bash
+
+function read_dir() {
+local dir=$1
+local file
+
+for file in $(ls "$dir"); do
+    if [ -d "$dir""/""$file" ]
+    then
+      read_dir "$dir""/""$file"
+    else
+      echo "$dir""/""$file"
+    fi
+done
+}
+
+
+read_dir renderedv1 |grep yaml  > rendered_file
+read_dir ../source-crs |grep yaml  > source_file
+
+
+while IFS= read -r file1; do
+    while IFS= read -r file2; do
+        if [ "${file1##*/}" = "${file2##*/}" ]
+        then
+            diff -u "$file1" "$file2"
```

Comment:
We need to capture whether this command returned non-zero so that if we detected any differences we can exit non-zero at the end of this script.